nexusutils/dataconverter/readers/xps/reader.py

```diff
@@ -0,0 +1,360 @@
+        if survey_count_ == 0 and count == 0:
+            count = count + 1
+            template[f"{root}@default"] = entry
+
+        binding_energy_coord = None
+        chan_count = "_chan"
+        for data_var in xr_data.data_vars:
+            scan = data_var
+            # Collecting only accumulated counts
+            # indivisual channeltron counts goes to detector data section
+            if chan_count not in scan:
+                key_indv_scn_dt = f"{modified_key}/{scan}"
+
+                key_indv_scn_dt_unit = f"{key_indv_scn_dt}/@units"
+
+                coord_nm = list(xr_data[data_var].coords)[0]
+                binding_energy_coord = np.array(xr_data[data_var][coord_nm])
+                template[key_indv_scn_dt] = xr_data[data_var].data
+                template[key_indv_scn_dt_unit] = config_dict[f"{key}/@units"]
+
+        key_data = f"{modified_key}/data"
+        key_data_unit = f"{key_data}/@units"
+
+        key_signal = f"{modified_key}/@signal"
+
+        be_nm = "BE"
+        be_index = 0
+        key_be = f"{modified_key}/{be_nm}"
+        key_be_unit = f"{key_be}/@units"
+        key_be_axes = f"{modified_key}/@axes"
+        key_be_ind = f"{modified_key}/@{be_nm}_indices"
+
+        # setting up AXISNAME
+        axisname = "AXISNAME[axisname]"
+        long_name = "Binding Energy"
+        key_ax_mn = f"{modified_key}/{axisname}"
+        key_ax_ln_nm = f"{modified_key}/{axisname}/@long_name"
+
+        key_nxclass = f"{modified_key}/@NX_class"
+
+        template[key_signal] = "data"
+        template[key_data] = np.mean([xr_data[x_arr].data
+                                     for x_arr in xr_data.data_vars
+                                     if "_chan" not in x_arr],
+                                     axis=0)
+
+        template[f"{key_data}_errors"] = \
+            np.std([xr_data[x_arr].data
+                    for x_arr in xr_data.data_vars
+                    if "_chan" not in x_arr], axis=0)
+        template[key_data_unit] = config_dict[f"{key}/@units"]
+        template[key_be_unit] = "eV"
+        template[key_be] = binding_energy_coord
+        template[key_be_axes] = be_nm
+        template[key_be_ind] = be_index
+        template[key_nxclass] = "NXdata"
+        template[key_ax_ln_nm] = long_name
+        template[key_ax_mn] = be_index
+
+
+def fill_detector_group(key,
+                        entries_values,
+                        config_dict,
+                        template,
+                        entry_set):
+    """Fill out fileds and attributes for NXdetector/NXdata"""
+
+    for entry, xr_data in entries_values.items():
+        entry_set.add(entry)
+
+        chan_count = "_chan"
+
+        # Iteration over scan
+        for data_var in xr_data.data_vars:
+
+            if chan_count in data_var:
+                detector_num = data_var.split("_chan")[-1]
+                detector_nm = f"detector{detector_num}"
+                DETECTOR_SET.add(detector_nm)
```

Comment:
The all uppercase notation is typically reserved for constants in python, i.e. they should not be changed here. Consider just specifying this as lower case.

Reply:
This gives pylint errors. That is the reason introducing module variables always uppercase.

Reply:
ok, my bad I thought that it is then seen as a constant. But obviously I'm wrong